test/mocks/mock_http_client.h

```diff
@@ -104,6 +104,11 @@ class MockHTTPClient : public IHTTPClient {
             std::map<std::string, std::string>()) {
         return this->GET_impl(url, headers);
     }
+    /**
+     */
+    virtual bool DL(const std::string& url, const std::string& path) {
```

Comment:
Actually, I think that you want this guy to be using GMock's stuff so that you can do call counting, test-based return values, etc:

```suggestion
    MOCK_METHOD2(DL, bool(const std::string& url, const std::string& path));
```

Reply:
Was hoping to come back to that stuff, but yes, absolutely.